lib/buyers/forms.py

```diff
@@ -12,10 +12,10 @@ def clean_pin(pin):
         return pin
 
     if not len(pin) == 4:
-        raise forms.ValidationError(PIN_4_NUMBERS_LONG)
+        raise forms.ValidationError(PIN_4_NUMBERS_LONG, code='invalid')
 
     if not pin.isdigit():
-        raise forms.ValidationError(PIN_ONLY_NUMBERS)
+        raise forms.ValidationError(PIN_ONLY_NUMBERS, code='invalid')
```

Comment:
oh I see. I guess it makes sense in the context of which field triggered the error. I was thinking it would be nice to use codes like this somewhere in the UI (recessed into the background) to compliment user errors so that QA reports would be useful at a glance. This approach has been really helpful for webpay.